native/cocos/renderer/pipeline/GlobalDescriptorSetManager.cpp

```diff
@@ -111,11 +112,12 @@ void GlobalDSManager::update() {
     }
 }
 
-gfx::DescriptorSet *GlobalDSManager::getOrCreateDescriptorSet(uint32_t idx) {
+gfx::DescriptorSet *GlobalDSManager::getOrCreateDescriptorSet(const scene::Light *light) {
+    CC_ASSERT(light);
     // The global descriptorSet is managed by the pipeline and binds the buffer
-    if (_descriptorSetMap.count(idx) <= 0 || (_descriptorSetMap.at(idx) == nullptr)) {
+    if (_descriptorSetMap.count(light) <= 0 || (_descriptorSetMap.at(light) == nullptr)) {
         auto *descriptorSet = _device->createDescriptorSet({_descriptorSetLayout});
-        _descriptorSetMap.emplace(idx, descriptorSet);
+        _descriptorSetMap.emplace(light, descriptorSet);
```

Comment:
Use `_descriptorSetMap[light] = descriptorSet `, as `emplace` will not replace the value if the key already exists.